src/reify.ml4

```diff
@@ -724,6 +911,58 @@ struct
       Command.declare_mutual_inductive_with_eliminations (mut_ind mr mf mp mi mpol mpr) [] [];()
     | _ -> raise (Failure "ill-typed mutual_inductive_entry")
 
+  let rec run_template_program_rec  ((env,evm,pgm): Environ.env * Evd.evar_map * Term.constr) : Environ.env * Evd.evar_map * Term.constr =
+    let (evm,pgm) = reduce_hnf env (evm, pgm) in 
+    let (coConstr,args) = app_full pgm [] in
+    if Term.eq_constr coConstr tmReturn then
+      match args with
+      | _::h::[] -> (env,evm,h)
+      | _ -> raise (Failure "tmReturn must take 2 arguments. Please file a bug with Template-Coq.")
```

Comment:
It would be nice to unify these error messages. Something like:

`raise (BadCommand pgm)`

where `pgm` is the `TemplateMonad`. At the toplevel, we can print it out.